Core/ELF/ElfReader.cpp

```diff
@@ -29,8 +29,8 @@ const char *ElfReader::GetSectionName(int section) const {
 		return nullptr;
 
 	int nameOffset = sections[section].sh_name;
-	if (nameOffset < 0) {  // TODO: Where can we get a solid upper limit?
-		ERROR_LOG(LOADER, "ELF: Bad name offset %d in section %d", nameOffset, section);
+	if (nameOffset < 0 || nameOffset >= size_) {
```

Comment:
Not sure what you mean by before size_, do you mean before nameOffset? And that would only take a byte load, not a memchr.